### PR TITLE
Add setting to hide NULL values on identify result

### PR DESCRIPTION
**Pull request by ismailsunni**
## Description

Based on #48746, it can be useful when you have a lot of null values.
Peek 2022-06-09 19-23 - hide null values on identify result

Funded by Camptocamp https://www.camptocamp.com/

### CODE REVIEW

File: src/app/qgsidentifyresultsdialog.cpp [hunk 1]
```diff
@@ -709,6 +712,10 @@ QgsIdentifyResultsFeatureItem *QgsIdentifyResultsDialog::createFeatureItem( QgsV
       defVal = vlayer->dataProvider()->defaultValueClause( fields.fieldOriginIndex( i ) );
 
     const QString originalValue = defVal == attrs.at( i ) ? defVal : fields.at( i ).displayString( attrs.at( i ) );
+    if ( originalValue.size() == 0 && QgsSettings().value( QStringLiteral( "/Map/hideNullValues" ), false ).toBool() )
```
Comment (nyalldawson) [inline src/app/qgsidentifyresultsdialog.cpp:715]: I don't think @3nids wants anymore raw QgsSettings use in the code anymore -- this should instead be ported to the new settings framework. See e.g. qgssettingsregistryapp.cpp

Comment (nyalldawson) [inline src/app/qgsidentifyresultsdialog.cpp:715]: Checking for an empty string != a null value. This should instead be something like `if ( attrs.at( i ).isNull() )`

**Comment by ismailsunni** [comment]
Thanks, @nyalldawson , I have addressed your review

File: src/app/qgsidentifyresultsdialog.cpp [hunk 2]
```diff
@@ -444,6 +445,8 @@ QgsIdentifyResultsDialog::QgsIdentifyResultsDialog( QgsMapCanvas *canvas, QWidge
   mActionAutoFeatureForm->setChecked( mySettings.value( QStringLiteral( "Map/identifyAutoFeatureForm" ), false ).toBool() );
   settingsMenu->addAction( mActionHideDerivedAttributes );
   mActionHideDerivedAttributes->setChecked( mySettings.value( QStringLiteral( "Map/hideDerivedAttributes" ), false ).toBool() );
+  settingsMenu->addAction( mActionHideNullValues );
+  mActionHideNullValues->setChecked( mySettings.value( QStringLiteral( "Map/hideNullValues" ), false ).toBool() );
```
Comment (nyalldawson) [inline src/app/qgsidentifyresultsdialog.cpp:449]:
```suggestion
  mActionHideNullValues->setChecked( QgsIdentifyResultsDialog::settingHideNullValues.value() );
```

File: src/app/qgsidentifyresultsdialog.h [hunk 3]
```diff
@@ -133,6 +134,8 @@ class APP_EXPORT QgsIdentifyResultsDialog: public QDialog, private Ui::QgsIdenti
 
     ~QgsIdentifyResultsDialog() override;
 
+    static const inline QgsSettingsEntryBool settingHideNullValues = QgsSettingsEntryBool( QStringLiteral( "hideNullValues" ), QgsSettings::Prefix::MAP, false, QStringLiteral( "Whether hide or show attribute with NULL values in the identify feature result" ) );
```
Comment (nyalldawson) [inline src/app/qgsidentifyresultsdialog.h:137]:
```suggestion
    static const inline QgsSettingsEntryBool settingHideNullValues = QgsSettingsEntryBool( QStringLiteral( "hideNullValues" ), QgsSettings::Prefix::MAP, false, QStringLiteral( "Whether to hide attributes with NULL values in the identify feature result" ) );
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**